### PR TITLE
Separate Remote State and Publication enabled and configured methods

### PR DESCRIPTION
**Pull request by shiv0408**
### Description
Currently we have references to `isRemoteStateEnabled` and `isRemotePublicationEnabled` methods at multiple places. We want to create different methods `isRemoteStateConfigured` and `isRemotePublicationConfigured`, which will return true if node attributes are present on the node. 

### Check List
- [x] Functionality includes testing.
- [ ] API changes companion pull request [created](https://github.com/opensearch-project/opensearch-api-specification/blob/main/DEVELOPER_GUIDE.md), if applicable.
- [ ] Public documentation issue/PR [created](https://github.com/opensearch-project/documentation-website/issues/new/choose), if applicable.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: server/src/main/java/org/opensearch/node/Node.java [hunk 1]
```diff
@@ -796,7 +796,7 @@ protected Node(
             final RemoteClusterStateService remoteClusterStateService;
             final RemoteClusterStateCleanupManager remoteClusterStateCleanupManager;
             final RemoteIndexPathUploader remoteIndexPathUploader;
-            if (isRemoteStoreClusterStateEnabled(settings)) {
+            if (isRemoteClusterStateConfigured(settings)) {
```
[the remark this answers is not in the document]
Comment (shiv0408) [reply server/src/main/java/org/opensearch/node/Node.java:799]: Segment and translog repositories should not have any effect here. Can you share some case where initializing can create any issue?

[the remark this answers is not in the document]
Comment (rajiv-kv) [reply server/src/main/java/org/opensearch/node/Node.java:799]: Lets says the node has only these three repo's set
- cluster.state
- cluster.segment
- cluster.translog

The routing-table repository is not set. Would the RemoteService get initialized ?

[the remark this answers is not in the document]
Comment (shiv0408) [reply server/src/main/java/org/opensearch/node/Node.java:799]: As cluster state repo is present, we would still be initializing the RemoteClusterStateService and other related services. With services initialized, we would only be using the methods from these services only if setting is enabled on the nodes.
`remote_state.enabled: false` : No interaction with any service while publish phase.
`remote_state.enabled: true & remote_publication.enabled: false` : Upload metadata related objects to remote store, but transport calls are still there for publication
`remote_state.enabled: true & remote_publication.enabled: true` : full cluster state is uploaded to remote store and path to manifest is sent to all the nodes during publish.

File: server/src/main/java/org/opensearch/gateway/remote/RemoteClusterStateService.java [hunk 2]
```diff
@@ -256,7 +256,7 @@ public RemoteClusterStateService(
         List<IndexMetadataUploadListener> indexMetadataUploadListeners,
         NamedWriteableRegistry namedWriteableRegistry
     ) {
-        assert isRemoteStoreClusterStateEnabled(settings) : "Remote cluster state is not enabled";
+        assert isRemoteClusterStateConfigured(settings) : "Remote cluster state is not enabled";
```
Comment (rajiv-kv) [inline server/src/main/java/org/opensearch/gateway/remote/RemoteClusterStateService.java:259]: nit: update the message `is not configured`

File: server/src/internalClusterTest/java/org/opensearch/gateway/remote/RemoteStatePublicationIT.java [hunk 3]
```diff
@@ -215,7 +215,11 @@ public void testRemotePublicationDisableIfRemoteStateDisabled() {
         ensureStableCluster(5);
         ensureGreen(INDEX_NAME);
 
-        assertNull(internalCluster().getCurrentClusterManagerNodeInstance(RemoteClusterStateService.class));
+        RemoteClusterStateService remoteClusterStateService = internalCluster().getCurrentClusterManagerNodeInstance(
```
[the remark this answers is not in the document]
Comment (shiv0408) [reply server/src/internalClusterTest/java/org/opensearch/gateway/remote/RemoteStatePublicationIT.java:218]: Added in latest commit

**Review by rajiv-kv — APPROVED** [review]

**Review by shwetathareja — APPROVED** [review]

**Pull request merged**